### PR TITLE
Fixes needed for adding a node into existing cluster

### PR DESCRIPTION
**Pull request by jsuchome**
These are various bits found during testing the case of new node being added to existing pacemaker cluster (bsc#1083427)

- At least in the special case of adding new member to existing cluster,  Openstack::HA.set_controller_role is called and node[:pacemaker] accessed.
- cinder: Do not access role attributes with symbol keys 
- neutron: Check network presence before accessing it

### CODE REVIEW

**Comment by jsuchome** [comment]
Hi guys, any chance for another review?

File: crowbar_framework/app/models/cinder_service.rb [hunk 1]
```diff
@@ -243,14 +243,13 @@ def apply_role_pre_chef_call(old_role, role, all_nodes)
...
     proposal = Proposal.find_by(barclamp: "cinder", name: role.inst)
-    role.default_attributes[:cinder][:volumes].each_with_index do |volume, volid|
-      next unless volume[:backend_driver] == "rbd"
-      if volume[:rbd][:secret_uuid].empty?
-        secret_uuid = `uuidgen`.strip
-        volume[:rbd][:secret_uuid] = secret_uuid
-        proposal[:attributes][:cinder][:volumes][volid][:rbd][:secret_uuid] = secret_uuid
-        dirty = true
-      end
+    role.default_attributes["cinder"]["volumes"].each_with_index do |volume, volid|
```
Comment (Itxaka) [inline crowbar_framework/app/models/cinder_service.rb]: any reason of changing the sym to string here? AFAIK it should be ok to get it with either as this was patched on our chef server

Comment (jsuchome) [reply crowbar_framework/app/models/cinder_service.rb]: @Itxaka I do not exactly know why is it required, only that it fails with symbols. We're doing strings in other places as well (models accessing role attributes).

Now, when you're mentioning it, it's possible it works with symbols in master, I've tested this on 4.0 actually

Comment (Itxaka) [reply crowbar_framework/app/models/cinder_service.rb]: :+1:

**Pull request merged**